uwblueprint/focus-on-nature
# feat: add image upload

## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Tasks-38b49f83f6ce4b07ae188dabb37631e8?p=1b2a2134dfa84cb1b461fea7a58a70a8)



## Implementation description
* added image upload middleware in `POST /camp` route, where image and camp data should be sent in `form-data`. `form-data` contains 2 keys: file containing the image that needs to be uploaded, and body containing camp data that needs to be saved. When image is successfully uploaded, a key: `fileName` containing image name gets stored in db along with the camp data
* "file" key in form-data is optional i.e. uploading an image in `POST /camp` route is optional. If the image is not provided, then `fileName` key will not be created in db
* max image upload size is 5MB


## Steps to test
1. In order to test the changes, go to postman->body->form-data. Create 2 keys: file containing the image that needs to be uploaded, and body containing the camp data that needs to be stored in db. You can use the below sample body input:
`{"active": true, "ageLower": 15, "ageUpper": 30, "name": "test-camp-image", "description": "image-test-description", "location": "canada", "capacity": 20,  "fee": 25,  "formQuestions": [{"question": "Q1", "type": "Text", "required": true, "description": "question description"}], "campSessions": []}`

![Screenshot 2022-04-13 at 4 28 25 PM](https://user-images.githubusercontent.com/67038446/163280449-83a074c4-92f8-44d4-a359-92f750d19ecd.png)

In order to verify if correct image is uploaded, you can refer to this [link](https://console.firebase.google.com/u/6/project/focus-on-nature-6c0b1/storage/focus-on-nature-6c0b1.appspot.com/files)



## What should reviewers focus on?
* does `fileName` key needs to be created if there is no image to be uploaded?

## Checklist
